tools/RestoreCommitterDateRoughly.ps1

```diff
@@ -0,0 +1,27 @@
+﻿<#
+	.SYNOPSIS
+	Git ワーキングツリーに含まれるファイルの更新日時に最終コミット日時を設定します。
+
+	.DESCRIPTION
+	Git でチェックアウトしたファイルの更新日時はチェックアウト日時になります。これはファイルのタイムスタンプを比較する場合に不都合な場合があり、代わりに最終コミット日時をタイムスタンプに設定します。
+
+	Git リポジトリ内で実行します。すべての階層のファイルが対象になります。
+
+	制限: ログを遡りすべてのファイルの最終コミット日時を取得することは１分以上かかることがあるため、このスクリプトでは１か月を上限としてログを遡ります。そのため最も古いタイムスタンプは実行日時を基準として前月同日の00:00:00になります。
+#>
+$TopDir = git rev-parse --show-toplevel
+$Oldest = (Get-Date).AddMonths(-1) # Of course, this is incorrect for the oldest timestamp. That's what "Roughly" means.
+
+$t = $Oldest
+@(
+	"?$($Oldest.ToString("yyyy-MM-dd 00:00:00"))"
+	, (git ls-files --full-name "$TopDir")
+	, (git log --format=format:?%ci --name-only --since=$($Oldest.ToString("yyyy-MM-dd")) --reverse)
```

Comment:
`--since=$($Oldest.ToString("yyyy-MM-dd"))` は単に `--since="1 month ago"` とすればいいと思います。

参考: https://stackoverflow.com/questions/14618022/how-does-git-log-since-count

Reply:
そう書けると言うことは知りませんでした。それによって $Oldest が省けるならそうしようと思いましたがどうもそうではありませんでした。その場合 `--since-"1 month ago"` の使用は「そうすることもできる」ではなく「そうすべきではない」ものになると思います。「１か月」というキーワードがスクリプト中の２か所に現れることになるからです。

Reply:
論点を勘違いしていました。時分秒を付けます。